### ISSUE TITLE
TypeError: __init__() got an unexpected keyword argument 'disable_buffering'

### ISSUE THREAD

**Issue by stam** [docker/compose#8499]
## Description of the issue
docker-compose up fails to run as it gives a TypeError:

## Context information (for bug reports)
Ubuntu 20.04, tried on python 3.8 and python 3.6.
**Output of `docker-compose version`**
```
docker-compose version 1.29.2, build unknown
docker-py version: 5.0.1
CPython version: 3.6.14
OpenSSL version: OpenSSL 1.1.1f  31 Mar 2020
```
/!\ If `docker-compose version` reports v2.0...` you're using the wrong repository, please report issues on [Compose-CLI](https://github.com/docker/compose-cli/issues/new)

**Output of `docker version`**
```
Client: Docker Engine - Community
 Version:           20.10.8
 API version:       1.41
 Go version:        go1.16.6
 Git commit:        3967b7d
 Built:             Fri Jul 30 19:54:08 2021
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.8
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.16.6
  Git commit:       75249d8
  Built:            Fri Jul 30 19:52:16 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.4.9
  GitCommit:        e25210fe30a0a703442421b0f60afac609f950a3
 runc:
  Version:          1.0.1
  GitCommit:        v1.0.1-0-g4144b63
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

**Output of `docker-compose config`**
(Make sure to add the relevant `-f` and other flags)
```
services:
  backend:
    container_name: kg-demo-api
    environment:
      DB_URL: SECRET
    image: SOME_URL/backend:latest
    volumes:
    - /data/nginx/www/static:/static:rw
  frontend:
    container_name: kg-demo-ui
    image: SOME_URL/frontend:latest
    restart: unless-stopped
  nginx:
    container_name: nginx
    image: SOME_URL/nginx:latest
    ports:
    - published: 80
      target: 80
    - published: 443
      target: 443
    restart: unless-stopped
    volumes:
    - /data/nginx/conf.d:/etc/nginx/conf.d:rw
    - /etc/letsencrypt:/etc/letsencrypt:rw
    - /data/nginx/www:/var/www:rw
version: '3'

```

## Steps to reproduce the issue

1. Run docker-compose up 
2.
3.

### Observed result
The specific backend docker container is not running, but the other 2 are. 

### Expected result
Everything running or at least dumping stacktraces

### Stacktrace / full error message

```
nginx is up-to-date
Starting kg-demo-api ...
kg-demo-ui is up-to-date

ERROR: for kg-demo-api  __init__() got an unexpected keyword argument 'disable_buffering'

ERROR: for backend  __init__() got an unexpected keyword argument 'disable_buffering'
Traceback (most recent call last):
  File "/usr/local/bin/docker-compose", line 8, in <module>
    sys.exit(main())
  File "/home/borg/.local/lib/python3.6/site-packages/compose/cli/main.py", line 81, in main
    command_func()
  File "/home/borg/.local/lib/python3.6/site-packages/compose/cli/main.py", line 203, in perform_command
    handler(command, command_options)
  File "/home/borg/.local/lib/python3.6/site-packages/compose/metrics/decorator.py", line 18, in wrapper
    result = fn(*args, **kwargs)
  File "/home/borg/.local/lib/python3.6/site-packages/compose/cli/main.py", line 1186, in up
    to_attach = up(False)
  File "/home/borg/.local/lib/python3.6/site-packages/compose/cli/main.py", line 1182, in up
    attach_dependencies=attach_dependencies,
  File "/home/borg/.local/lib/python3.6/site-packages/compose/project.py", line 702, in up
    get_deps,
  File "/home/borg/.local/lib/python3.6/site-packages/compose/parallel.py", line 108, in parallel_execute
    raise error_to_reraise
  File "/home/borg/.local/lib/python3.6/site-packages/compose/parallel.py", line 206, in producer
    result = func(obj)
  File "/home/borg/.local/lib/python3.6/site-packages/compose/project.py", line 688, in do
    override_options=override_options,
  File "/home/borg/.local/lib/python3.6/site-packages/compose/service.py", line 586, in execute_convergence_plan
    containers, scale, timeout, detached, start
  File "/home/borg/.local/lib/python3.6/site-packages/compose/service.py", line 524, in _execute_convergence_start
    "Starting",
  File "/home/borg/.local/lib/python3.6/site-packages/compose/parallel.py", line 108, in parallel_execute
    raise error_to_reraise
  File "/home/borg/.local/lib/python3.6/site-packages/compose/parallel.py", line 206, in producer
    result = func(obj)
  File "/home/borg/.local/lib/python3.6/site-packages/compose/service.py", line 522, in <lambda>
    lambda c: self.start_container_if_stopped(c, attach_logs=not detached, quiet=True),
  File "/home/borg/.local/lib/python3.6/site-packages/compose/service.py", line 637, in start_container_if_stopped
    container.attach_log_stream()
  File "/home/borg/.local/lib/python3.6/site-packages/compose/container.py", line 202, in attach_log_stream
    self.log_stream = self.attach(stdout=True, stderr=True, stream=True)
  File "/home/borg/.local/lib/python3.6/site-packages/compose/container.py", line 294, in attach
    return self.client.attach(self.id, *args, **kwargs)
  File "/home/borg/.local/lib/python3.6/site-packages/docker/utils/decorators.py", line 19, in wrapped
    return f(self, resource_id, *args, **kwargs)
  File "/home/borg/.local/lib/python3.6/site-packages/docker/api/container.py", line 58, in attach
    response = self._post(u, headers=headers, params=params, stream=True)
  File "/home/borg/.local/lib/python3.6/site-packages/docker/utils/decorators.py", line 46, in inner
    return f(self, *args, **kwargs)
  File "/home/borg/.local/lib/python3.6/site-packages/docker/api/client.py", line 233, in _post
    return self.post(url, **self._set_request_timeout(kwargs))
  File "/usr/lib/python3/dist-packages/requests/sessions.py", line 581, in post
    return self.request('POST', url, data=data, json=json, **kwargs)
  File "/usr/lib/python3/dist-packages/requests/sessions.py", line 533, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/lib/python3/dist-packages/requests/sessions.py", line 646, in send
    r = adapter.send(request, **kwargs)
  File "/usr/lib/python3/dist-packages/requests/adapters.py", line 449, in send
    timeout=timeout
  File "/usr/lib/python3/dist-packages/urllib3/connectionpool.py", line 672, in urlopen
    chunked=chunked,
  File "/usr/lib/python3/dist-packages/urllib3/connectionpool.py", line 421, in _make_request
    six.raise_from(e, None)
  File "<string>", line 3, in raise_from
  File "/usr/lib/python3/dist-packages/urllib3/connectionpool.py", line 416, in _make_request
    httplib_response = conn.getresponse()
  File "/usr/lib/python3.6/http/client.py", line 1379, in getresponse
    response = self.response_class(self.sock, method=self._method)
  File "/home/borg/.local/lib/python3.6/site-packages/docker/transport/unixconn.py", line 43, in response_class
    return httplib.HTTPResponse(sock, *args, **kwargs)
TypeError: __init__() got an unexpected keyword argument 'disable_buffering'
```

## Additional information

OS version / distribution, `docker-compose` install method, etc.
Ubuntu 20.04, python 3.6 (also tried python 3.8) everything installed with ansible scripts, https://github.com/docker/compose/releases/download/1.27.4/docker-compose-Linux-x86_64 and with the `pip install docker-compose`

**Comment by AlexBaily** [comment]
Getting the same with Ubuntu 20.04.3 and Python 3.8. Only happening on some images as well, those being node:8.11-alpine.

**Comment by Aposhian** [comment]
This is caused by `docker-py` version 5.0.1. A fix has been merged, but a new release is pending. Workaround is to pin `docker-py` to 5.0.0. https://github.com/docker/docker-py/issues/2885

**Comment by KenwoodFox** [comment]
> This is caused by `docker-py` version 5.0.1. A fix has been merged, but a new release is pending. Workaround is to pin `docker-py` to 5.0.0. [docker/docker-py#2885](https://github.com/docker/docker-py/issues/2885)

Installing `docker-py` results in a new error
```
Traceback (most recent call last):
  File "/usr/local/bin/docker-compose", line 6, in <module>
    from compose.cli.main import main
  File "/usr/local/lib/python3.7/dist-packages/compose/cli/main.py", line 29, in <module>
    from ..metrics.decorator import metrics
  File "/usr/local/lib/python3.7/dist-packages/compose/metrics/decorator.py", line 3, in <module>
    from compose.metrics.client import MetricsCommand
  File "/usr/local/lib/python3.7/dist-packages/compose/metrics/client.py", line 5, in <module>
    from docker import ContextAPI
ImportError: cannot import name 'ContextAPI' from 'docker' (/home/vestabot/.local/lib/python3.7/site-packages/docker/__init__.py)
```

Without docker-py i get the __init__() error.

**Comment by Aposhian** [comment]
This should be closed for good after #9401 is merged
